src/modules/status-list/status-list.controller.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpStatus,
+  Param,
+  Post,
+  Res,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+import { Response } from 'express';
+import { User } from '../../common/user.decorator';
+import { RoleService } from '../role/role.service';
+import { CreateEntryInputDto } from './dtos/create-entry-input.dto';
+import { CredentialWithStatusDto } from './dtos/credential-status.dto';
+import { RegisterRevokeInputDto } from './dtos/register-revoke-input.dto';
+import { SignRevokeInputDto } from './dtos/sign-revoke-input.dto';
+import { StatusListCredentialDto } from './dtos/status-list-credential.dto';
+import { StatusListVerifiableCredentialDto } from './dtos/status-list-verifiable-credential.dto';
+import { STATUS_LIST_MODULE_PATH } from './status-list.const';
+import { StatusListService } from './status-list.service';
+
+@Controller({ path: STATUS_LIST_MODULE_PATH, version: '1' })
+@ApiTags('status-list')
+export class StatusListController {
+  constructor(
+    private readonly statusListService: StatusListService,
+    private readonly roleService: RoleService
+  ) {}
+
+  @Post('/create-entry')
+  @ApiResponse({ type: CredentialWithStatusDto })
+  async createEntry(
+    @User() currentUser: string,
+    @Body() { credential }: CreateEntryInputDto
+  ): Promise<CredentialWithStatusDto> {
+    const namespace = credential.credentialSubject.role.namespace;
+    const issuer = credential.issuer;
+
+    if (issuer !== currentUser) {
+      throw new UnauthorizedException(
+        'You are not an issuer of given credential'
+      );
+    }
+
+    await this.roleService.verifyEnrolmentIssuer({
+      issuerDID: issuer,
+      claimType: namespace,
+    });
+
+    return await this.statusListService.createEntry(credential);
+  }
+
+  @Post('/register-revoke')
+  async registerRevoke(
```

Comment:
We're straying from VC-API due to our client-side keys, but I think that this path is a bit more elegant and communicates that we are keeping VC-API in mind https://w3c-ccg.github.io/vc-api/#update-status
```suggestion
  @Post('/credentials/update/register')
  async registerStatusUpdate(
```

Reply:
Actually, maybe this is better
```suggestion
  @Post('/credentials/update/initiate')
  async initiateStatusUpdate(
```